recipes/libdatachannel/all/conanfile.py

```diff
@@ -0,0 +1,138 @@
+        tc.cache_variables["USE_NICE"] = self.options.with_nice
+        tc.cache_variables["USE_SYSTEM_JUICE"] = not self.options.with_nice
+        if self.options.with_ssl == "gnutls":
+            tc.cache_variables["USE_GNUTLS"] = True
+        elif self.options.with_ssl == "mbedtls":
+            tc.cache_variables["USE_MBEDTLS"] = True
+        tc.generate()
+        deps = CMakeDeps(self)
+        deps.set_property("usrsctp", "cmake_target_name", "Usrsctp::Usrsctp")
+        deps.set_property("libsrtp", "cmake_file_name", "libSRTP")
+        deps.set_property("libsrtp", "cmake_target_name", "libSRTP::srtp2")
+        if self.options.with_ssl == "mbedtls":
+            deps.set_property("mbedtls::libembedtls", "cmake_target_name", "MbedTLS::MbedTLS")
+        elif self.options.with_ssl == "gnutls" and self.options.with_websocket:
+            deps.set_property("nettle", "cmake_file_name", "Nettle")
+            deps.set_property("nettle", "cmake_target_name", "Nettle::Nettle")
+        if not self.options.with_nice and not self.options.shared:
+            # Targetname is LibJuice::LibJuiceStatic for static, but upstream makes no distinction
+            deps.set_property("libjuice", "cmake_target_name", "LibJuice::LibJuice")
+        deps.generate()
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        copy(self, "LICENSE", self.source_folder, os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
+        cmake.install()
+        fix_apple_shared_install_name(self)
+
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
+        rm(self, "*.la", os.path.join(self.package_folder, "lib"))
+        rm(self, "*.pdb", os.path.join(self.package_folder, "lib"))
+        rm(self, "*.pdb", os.path.join(self.package_folder, "bin"))
+
+    def package_info(self):
+        suffix = ""
+        if is_msvc(self) and self.settings.build_type == "Debug":
+            suffix = "d"
+        self.cpp_info.libs = ["datachannel" + suffix]
+        self.cpp_info.set_property("cmake_file_name", "LibDataChannel")
+        self.cpp_info.set_property("cmake_target_name", "LibDataChannel::LibDataChannel")
+
+        if self.settings.os in ["Linux", "FreeBSD"]:
+            self.cpp_info.system_libs.extend(["m", "pthread", "dl"])
+
+        if not self.options.shared:
+            self.cpp_info.defines.append("RTC_STATIC")
+
+        if self.settings.os == "Windows":
+            self.cpp_info.system_libs.append("ws2_32")
+            self.cpp_info.defines.append("WIN32_LEAN_AND_MEAN")
+
+        if is_msvc(self):
+            self.cpp_info.cxxflags.append("/bigobj")
+            self.cpp_info.defines.extend(["_CRT_SECURE_NO_WARNINGS", "NOMINMAX"])
```

Comment:
@AbrilRBS - why are these being defined publicly for consumers in `package_info()`?
 These are not defined publicly in the upstream project, they only apply to the project itself - and generally these should never be defined "publicly" for consumers. These will affect all translation units in consumers, and those macros controls behaviours where the consumers should have full control - we shouldn't disable security warnings on _their_ code.

For what is worth, what I see the upstream project defining is:
```
  INTERFACE_COMPILE_DEFINITIONS "RTC_STATIC;RTC_ENABLE_WEBSOCKET=1;RTC_ENABLE_MEDIA=1"
```

and on Windows (the same ones)
```
  INTERFACE_COMPILE_DEFINITIONS "RTC_STATIC;RTC_ENABLE_WEBSOCKET=1;RTC_ENABLE_MEDIA=1"
```

I can't find any reference or justification for transitively defining any of the below 
* `WIN32_LEAN_AND_MEAN`
* `/bigobj`
* `_CRT_SECURE_NO_WARNINGS`
* `NOMINMAX`

Am I missing something here?

Reply:
No, it was indeed me who was missing some defines, and proper libnice cmake names